server/app/query_visualization_artifact_execution.py

```diff
@@ -0,0 +1,165 @@
+from cmflib import cmfquery
+from collections import deque, defaultdict
+import warnings
+
+warnings.filterwarnings("ignore")
+
+class UniqueQueue:
+    def __init__(self):
+        self.queue = deque()
+        self.seen = set()
+    
+    def enqueue(self, value):
+        if value not in self.seen:
+            self.queue.append(value)
+            self.seen.add(value)
+    
+    def dequeue(self):
+        if self.queue:
+            value = self.queue.popleft()
+            self.seen.remove(value)
+            return value
+        raise IndexError("dequeue from an empty queue")
+    
+    def __len__(self):
+        return len(self.queue)
+    
+    def __contains__(self, value):
+        return value in self.seen
```

Comment:
Is this class used anywhere? If yes, we can move this to some helper/util folder in separate file. We can add a doc string explaining functionality of the class .

Reply:
Hi Shreyas, this is redundant code that we are not using, and we have now fixed it in the code. Thank you for catching that!